src/main.rs

```diff
@@ -30,14 +32,57 @@ fn main() {
     let vm = VirtualMachine::new(settings);
 
     let res = run_rustpython(&vm, &matches);
-    // See if any exception leaked out:
-    handle_exception(&vm, res);
 
     #[cfg(feature = "flame-it")]
     {
         main_guard.end();
         if let Err(e) = write_profile(&matches) {
             error!("Error writing profile information: {}", e);
+        }
+    }
+
+    // See if any exception leaked out:
+    if let Err(err) = res {
+        if objtype::isinstance(&err, &vm.ctx.exceptions.system_exit) {
+            let args = vm.get_attribute(err.clone(), "args").unwrap();
+            let args = args.payload::<PyTuple>().expect("'args' must be a tuple");
+            match args.elements.len() {
+                0 => return,
+                1 => match_class!(match args.elements[0].clone() {
+                    i @ PyInt => {
+                        use num_traits::cast::ToPrimitive;
+                        std::process::exit(i.as_bigint().to_i32().unwrap());
+                    }
+                    _ => {
+                        if vm.is_none(&args.elements[0]) {
+                            return;
+                        }
+                        let s = vm.to_str(&args.elements[0]);
+                        let s = s
+                            .as_ref()
+                            .ok()
+                            .map_or("<element str() failed>", |s| s.as_str());
+                        println!("{}", s);
+                    }
+                }),
+                _ => {
```

Comment:
is this arm same to `print!("{}", vm.to_str(&args))`?

Reply:
Yep, I think I was trying to emulate code from `exceptions.rs` but it's just the `str` of it I guess.

Reply:
Actually, it's the repr of the args tuple.